lib/ios/app.nit

```diff
@@ -0,0 +1,187 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Basic structure for Nit apps on iOS
+module app
+
+import platform
+import ::app
+
+in "ObjC Header" `{
+	#import <UIKit/UIKit.h>
+
+	// Our interface to the iOS system
+	@interface AppDelegate: UIResponder <UIApplicationDelegate>
+
+		// The main window
+		@property (strong, nonatomic) UIWindow *window;
+	@end
+`}
+
+in "ObjC" `{
+
+	// Global reference to the App from app.nit
+	App app_nit_ios_app;
+
+	// Our own C argc and argv
+	int app_nit_ios_argc;
+	char **app_nit_ios_argv;
+
+	@implementation AppDelegate
+
+	- (BOOL)application:(UIApplication *)application
+		willFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+
+		// Set aside `application` to be used from Nit
+		App_ui_application__assign(app_nit_ios_app, application);
+		App_app_delegate__assign(app_nit_ios_app, self);
+
+		// Complete the callback
+		return App_will_finish_launching_with_options(app_nit_ios_app);
+	}
+
+	- (BOOL)application:(UIApplication *)application
+		didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+
+		return App_did_finish_launching_with_options(app_nit_ios_app);
+	}
+
+	- (void)applicationWillResignActive:(UIApplication *)application {
+		App_will_resign_active(app_nit_ios_app);
+	}
+
+	- (void)applicationDidEnterBackground:(UIApplication *)application {
+		App_did_enter_background(app_nit_ios_app);
+	}
+
+	- (void)applicationWillEnterForeground:(UIApplication *)application {
+		App_will_enter_foreground(app_nit_ios_app);
+	}
+
+	- (void)applicationDidBecomeActive:(UIApplication *)application {
+		App_did_become_active(app_nit_ios_app);
+	}
+
+	- (void)applicationWillTerminate:(UIApplication *)application {
+		App_will_terminate(app_nit_ios_app);
+	}
+
+	@end
+`}
+
+# Application interface to the iOS system
+extern class AppDelegate in "ObjC" `{ AppDelegate * `}
+end
+
+# Graphical application to which events are sent
+extern class UIApplication in "ObjC" `{ UIApplication * `}
+end
```

Comment:
what is the point of these two empty classes? I mean, if the class is empty then for what can they be useful? Maybe extend the comment to answer this mystery.